ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -0,0 +1,872 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2017-2018, Linaro Limited
+ */
+
+#ifndef __SKS_TA_H__
+#define __SKS_TA_H__
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <util.h>
+
+#define TA_SKS_UUID { 0xfd02c9da, 0x306c, 0x48c7, \
+			{ 0xa4, 0x9c, 0xbb, 0xd8, 0x27, 0xae, 0x86, 0xee } }
+
+/* SKS trusted application version information */
+#define SKS_VERSION_ID0		0
+#define SKS_VERSION_ID1		0
+
+/*
+ * SKS trusted application may receive requests requesting a number of
+ * parameters that may not suit the GPD 4 parameters directives.
+ * To factorize the use of GPD parameters for SKS services, the current
+ * API uses the 4 GPD invocation parameters as following:
+ *
+ * - Parameter #0 is not used or is used as an input memory reference argument.
+ * The referred buffer hold the command directives arguments (handlers,
+ * ids, ...) expected by the command. When several arguments are expected,
+ * they are serialized with byte alignment in the single control buffer.
+ *
+ * When caller defines parameter #0 as an in/out memory reference, the
+ * parameter #0 is used to feedback caller with a detailed 32bit return
+ * code that allow to return finer error reporting to client than the GPD
+ * TEE API allows a TA to return. For this reason, the invocation commands
+ * below mark parameter #0 as "in(*)-memref" in the command description
+ * comments.
+ *
+ * - Parameter #1 is not used or is an input memory reference. It refers to
+ * the input data provided for the invoked service.
+ *
+ * - Parameter #2 is not used or is an output memory reference. It refers to
+ * the output buffer expected to be filled with the output data requested
+ * by the caller. These data can be a object handle, a ciphered stream, etc.
+ *
+ * - Parameter #3 is not used.
+ */
```

Comment:
I don't think this whole comment adds much. Assuming the parameters for each command are properly described, things should be clear enough.

Reply:
Ok, this may be confusing. 
It is a generic description that applies to all commands.